File: quiz-backend/src/main/java/pmn/models/Quiz.java
package pmn.models;

import javax.persistence.*;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serializable;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Quiz implements Serializable {

	@Id
	@GeneratedValue(strategy=GenerationType.IDENTITY)
	private Long id;
	private String name;
	private String description;
	private Boolean isShared;
	private Boolean isActive;

}

File: quiz-frontend/src/app/models/quiz.ts
export interface Quiz {
  id: number;
  name: string;
  description: string;
  isShared: boolean;
  isActive: boolean;
  nbOfQuestions: number;
}

export interface PageQuiz {
  products: Quiz[];
  page: number;
  size: number;
  totalPages: number;
}
